doc/source/Integration.rst

```diff
@@ -735,3 +737,61 @@ information:
+
+The obvious way to configure this feature is to include the following snippet in the simulation code:
+
+.. code-block:: c++
+
+  if (configure_fancy_feature) {
+    my_grackle_data->use_fancy_feature = 1;
+    my_grackle_data->fancy_feature_param = 5.0; // arbitrary value
+  }
+
+Inclusion of the above snippet, will prevent the simulation code from compiling if the user has a version of Grackle installed in which :c:data:`chemistry_data` does not have the ``use_fancy_feature`` and ``fancy_feature_param`` fields. Consequently, such users will to update Grackle.
+
+  * This can be inconvenient when a user has no interest in using this new feature, but needs an unrelated feature/bugfix introduced to the code in a subsequent changeset
+
+  * This is especially inconvenient if a user is prototying a new feature in a custom Grackle branch in which the :c:data:`chemistry_data` struct is missing these fields.
+
+The following snippet shows how the dynamic access API can be used in the same way for versions of Grackle that include these parameters, and don't set the features in cases 
+
+.. code-block:: c++
+
+  if (configure_fancy_feature) {
+    int* use_fancy_feature = local_chemistry_data_access_int(
+      my_grackle_data, "use_fancy_feature"
+    );
+    double* fancy_feature_param = local_chemistry_data_access_double(
+      my_grackle_data, "fancy_feature_param"
+    );
+
+    if ((use_fancy_feature == NULL) || (fancy_feature_param == NULL)){
+      fprintf(stderr, "Update grackle version to use fancy feature\n");
+    } else {
+      *use_fancy_feature = 1;
+      *fancy_feature_param = 5.0;
+    }
+  }
+
+There are a few points worth noting:
+
+  * As the above snippets show, the dynamic api clearly produces more verbose code when configuring :c:data:`chemistry_data` field-by-field. However, in codes where users configure Grackle by specifying the name of fields in the :c:data:`chemistry_data` struct and the associated values in a parameter file, the dynamic API can facillitate MUCH less verbose code. Under certain implementations, it may not even be necessary to modify a simulation code to support newly-introduced grackle parameters.
+
+  * The dynamic API is slower than configuring :c:data:`chemistry_data` in the classic approach. However, this shouldn't be an issue since :c:data:`chemistry_data` is usually just configured once when the simulation code starts up.
+
+  * The highlighted functions can also be used in tandem with other functions described in :ref:`dynamic_api_functions` to simplify (de)serialization of :c:data:`chemistry_data`
+
+  * The dynamic API does not currently provide a function for configuring string parameters. This currently just affects one member of :c:data:`chemistry_data`: :c:data:`grackle_data_file`.
```

Comment:
Is this still true?

Reply:
Good catch! That's no longer true -- I'll fix that